projects/gateway2/query/httproute.go

```diff
@@ -408,6 +421,24 @@ func (r *gatewayQueries) GetRoutesForGateway(ctx context.Context, gw *gwv1.Gatew
 	return ret, nil
 }
 
+func (r *gatewayQueries) wildcardNamespaceExists(ctx context.Context) (bool, error) {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: wellknown.RouteDelegationLabelSelectorWildcardNamespace,
+		},
+	}
+	err := r.client.Get(ctx, types.NamespacedName{Name: ns.Name}, ns)
```

Comment:
are we sure namespaces are in the controller runtime cache?

Reply:
yes we watch namespaces
```
func (c *controllerBuilder) watchNamespaces(_ context.Context) error {
	return ctrl.NewControllerManagedBy(c.cfg.Mgr).
		For(&corev1.Namespace{}).
		Complete(reconcile.Func(c.reconciler.ReconcileNamespaces))
}
```
```
	return run(ctx,
...
		controllerBuilder.watchNamespaces,
```
projects/gateway2/controller/controller.go